soh/soh/Enhancements/randomizer/randomizer.h

```diff
@@ -11,6 +12,7 @@ class Randomizer {
   private:
     std::unordered_map<RandomizerCheck, RandomizerGet> itemLocations;
     std::unordered_map<RandomizerCheck, std::string> hintLocations;
+    std::unordered_map<s32, s32> EntranceIndeces;
```

Comment:
I suppose by that logic though there's also no reason the `itemLocations` and `hintLocations` couldn't also be arrays, so on second though maybe it's best to leave this as is for consistency. You will want to swap out the s32's for the name of your enum type when/if you define that in `randomizerTypes.h` though.

Reply:
i think a map is better because that way we don't need to have every entrance in it. i can see how itemLocations might make more sense as an array though considering we populate it with all the vanilla items too

Reply:
I could be wrong, but doesn't 3drando put all the entrances in the spoilerfile regardless of settings and just have the ones that shouldn't be shuffled according to settings mapped the vanilla way? I know that's how it does it with the item locations. Though I suppose if we're playing a vanilla save this map would never get populated and thus wouldn't be using much memory, so the map probably is better either way.

Reply:
i'm not sure, but it seems like it'd be pretty silly to put every single entrance into the spoiler when only dungeon entrances are shuffled.

with items i know it's been every check from the get go, but i don't think that strategy has to be set in stone (maybe a good thing to try out sometime would be not putting skulls in if skullsanity is disabled)

Reply:
Yeah I'm now seeing that there are way more than 250 entrance indexes in the game so that should definitely be an unordered map so we don't keep array slots for entrances we don't need lol.